fem/nonlininteg.hpp

```diff
@@ -341,6 +354,56 @@ class VectorConvectionNLFIntegrator : public NonlinearFormIntegrator
    virtual void AddMultPA(const Vector &x, Vector &y) const;
 };
 
+/** @brief This class is used to assemble the skew-symmetric form of the
+nonlinear  term arising in the Navier Stokes equations
+\f$(u \cdot \nabla v, w )\f$*/
+class ConvectiveVectorConvectionNLFIntegrator : public NonlinearFormIntegrator
```

Comment:
If my understanding is correct, this class is equivalent to `VectorConvectionNLFIntegrator` but the gradient is computed differently, would it makes sense to inherit from `VectorConvectionNLFIntegrator`?
```suggestion
class ConvectiveVectorConvectionNLFIntegrator : public VectorConvectionNLFIntegrator
```
If this is the case you can also remove `GetRule` and the private attributes.

Reply:
I agree this makes more sense.